eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -0,0 +1,254 @@
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> indexes current configured database in configured search engine index.
+EOT
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->output = $output;
+        $bulkCount = $input->getArgument('bulk_count');
+
+        $output->writeln('Creating search index for the engine: ' . get_parent_class($this->searchHandler));
+
+        $this->searchHandler->purgeIndex();
+
+        if ($this->searchHandler instanceof ContentIndexing) {
+            $this->createContentIndex($bulkCount);
+        }
+
+        if ($this->searchHandler instanceof LocationIndexing) {
+            $this->createLocationsIndex($bulkCount);
+        }
+
+        $output->writeln(PHP_EOL . 'Finished creating search index for the engine: ' . get_parent_class($this->searchHandler));
+    }
+
+    /**
+     * Wrapper for indexing Content.
+     *
+     * @param int $bulkCount a number of object rows to fetch in single batch
+     */
+    private function createContentIndex($bulkCount)
+    {
+        $query = $this->databaseHandler->createSelectQuery();
+        $query->select('count(id)')
+            ->from('ezcontentobject')
+            ->where($query->expr->eq('status', ContentInfo::STATUS_PUBLISHED));
+        $stmt = $query->prepare();
+        $stmt->execute();
+        $totalCount = $stmt->fetchColumn();
+
+        $query = $this->databaseHandler->createSelectQuery();
+        $query->select('id', 'current_version')
+            ->from('ezcontentobject')
+            ->where($query->expr->eq('status', ContentInfo::STATUS_PUBLISHED));
+
+        $stmt = $query->prepare();
+        $stmt->execute();
+
+        $this->searchHandler->purgeIndex();
+
+        $this->output->writeln('Indexing Content...');
+
+        /** @var \Symfony\Component\Console\Helper\ProgressBar $progress */
+        $progress = new ProgressBar($this->output);
+        $progress->start($totalCount);
+        $i = 0;
+        do {
+            $contentObjects = [];
+
+            for ($k = 0; $k <= $bulkCount; ++$k) {
+                if (!$row = $stmt->fetch(PDO::FETCH_ASSOC)) {
+                    break;
+                }
+
+                try {
+                    $contentObjects[] = $this->persistenceHandler->contentHandler()->load(
+                        $row['id'],
+                        $row['current_version']
+                    );
+                } catch (NotFoundException $e) {
+                    $this->logWarning($progress, "Could not load current version of Content with id ${row['id']}, so skipped for indexing. Full exception: " . $e->getMessage());
+                }
+            }
+
+            foreach ($contentObjects as $contentObject) {
+                try {
+                    $this->searchHandler->indexContent($contentObject);
```

Comment:
> Good point, missed that issue. ProgressBar also needs to be passed in this case because 

sure, but handler should not have to know about this, so if we go this route we need a internal implementation of `LoggerInterface` as explained above.